domain/model/mesh_results.py

```diff
@@ -134,6 +134,16 @@ def incremental_update(self, src: ConnectionMatrix) -> None:
             self._connections[from_agent_id] = dst_row
         self.connection_timestamp_oldest, self.connection_timestamp_newest = self._get_timestamp_range()
 
+    def drop_samples_older_than(self, threshold: datetime) -> None:
+        for row in self._connections.values():
+            for conn in row.values():
+                for i, item in enumerate(conn.health):
+                    if item.timestamp < threshold:
+                        logger.debug("Dropping %d samples outside requested time window", len(conn.health) - i)
+                        conn.health = conn.health[:i]
```

Comment:
It is in general unsafe to modify a list while iterating it. This code works because the iteration stops after the modification. There is though no need to modify `conn.health` while iterating it. For example like this:
```
        for row in self._connections.values():
            for conn in row.values():
                 n = 0
                 while conn.health[-1].timestamp < theshold:
                     conn.health.pop()
                     n += 1
                 if n > 0:
                     logger.debug("Dropped %d samples older than %s", n, threshold.isoformat())
```
This approach is also little bit more efficient, because:
- it does not iterate over fresh items at the front,
- it does not have to rebuild the whole list on every truncation.